project/api.py

```diff
@@ -0,0 +1,204 @@
+from datetime import datetime, timedelta
+import flask
+from flask import request, jsonify
+from werkzeug.exceptions import HTTPException
+import pandas as pd
+
+
+app = flask.Flask(__name__)
+
+
+def to_milliseconds(dt):
+    """Convert datetime object to timestamp in millisecond format"""
+    return int(datetime.timestamp(dt)*1000.)
+
+
+def prepare_request(time_range_start, time_range_end):
+    """Convert datetime time range to  timestamp in millisecond format"""
+    datetime_start = datetime.strptime(time_range_start, '%Y-%m-%d %H:%M:%S')
+    datetime_end = datetime.strptime(time_range_end, '%Y-%m-%d %H:%M:%S')
+    datetime_start_code = to_milliseconds(datetime_start)
+    datetime_end_code = to_milliseconds(datetime_end)
+    return datetime_start_code, datetime_end_code
+
+
+@app.route('/', methods=['GET'])
+def home():
+    return "<h1>Junior Data Scientist/Data Engineer Test Task</h1>"
+
+
+@app.errorhandler(Exception)
```

Comment:
What is the purpose of `@app.errorhandler` decorator?

Reply:
It register a function to handle errors by code or exception class

Reply:
What are the all cases when this decorator will be called?

Reply:
When smth goes wrong except for two cases:
* When no data passed to request
* When data format were wrong